### PR TITLE
Allow setting of registration url via env variable

### PR DESCRIPTION
**Pull request by bellycat77**
Use env variable **NSM_REGISTER_AS_URL** for NSE registration https://github.com/networkservicemesh/cmd-nse-simple-vl3-docker/issues/119

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -340,11 +340,18 @@ func main() {
...
+	} else {
+		RegisterAsURL = &config.RegisterAsURL
+	}
+
 	nseRegistration := &registryapi.NetworkServiceEndpoint{
 		Name:                 config.Name,
 		NetworkServiceNames:  config.ServiceNames,
 		NetworkServiceLabels: make(map[string]*registryapi.NetworkServiceLabels),
-		Url:                  listenOn.String(),
+		Url:                  RegisterAsURL.String(),
```
Comment (denis-tingaikin) [inline main.go:354]:
```suggestion
		Url:                  listenOn.String(),
```

File: main.go [hunk 2]
```diff
@@ -340,11 +340,18 @@ func main() {
 		}
 	}
 
+	var RegisterAsURL *url.URL
+	if config.RegisterAsURL.String() == "" {
+		RegisterAsURL = listenOn
+	} else {
+		RegisterAsURL = &config.RegisterAsURL
+	}
```
Comment (denis-tingaikin) [inline main.go:348]:
```suggestion
	if  config.RegisterAsURL.String() == "" {
		listenOn = RegisterAsURL
	}
```

**Review by denis-tingaikin — COMMENTED** [review]
Added cosmetic comments.

Also note: we should use `RegisterAsURL` as listenOn if its presented. 

Attach docker test for covering the scenario OR logs from k8s running with set env.

File: main.go (hunk 1 above)
Comment (bellycat77) [reply main.go:354]: Discussed locally

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**